File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    answer = {}
    
    for p in participant:
        if p in answer:
            answer[p] += 1
        else:
            answer[p] = 1
    
    for c in completion:
        if c in answer:
            answer[c] -= 1
            
    # return answer {"leo":1,"kiki":0,"eden":0}
    for key, value in answer.items():
        if value == 1:
            return key
        
        

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        // participant 0 - completion X
        
        // 1. 모든 참가자 명단 
        Map<String, Integer> map = new HashMap<>();
        
        // 참가자를 key로, value는 1
        for(String p : participant) {
            map.put(p, map.getOrDefault(p, 0) + 1);
        }
        
        // completion 참가자 key, value는 0으로
        for(String c : completion) {
            map.put(c, map.get(c) - 1);
        }
        
        // map에서 value가 1인게 정답임
        for(String key : map.keySet()) {
            if(map.get(key) == 1) {
                return key;
            }
        }
        return "";
        
    }
}